From a27ca01155d0d498b497054457031c5b0ca34abe Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 6 Apr 2016 17:51:49 +0300
Subject: [PATCH 1009/1240] dts: add customer device tree for Armada-70x0

- issue found when disable devel board config the U-Boot will not compile

Change-Id: I931de8bbcb626ea16464858175441d8e29a9d39a
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28835
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 arch/arm/dts/Makefile                  |   1 +
 arch/arm/dts/armada-70x0-customer0.dts | 175 +++++++++++++++++++++++++++++++++
 2 files changed, 176 insertions(+)
 create mode 100644 arch/arm/dts/armada-70x0-customer0.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 6109000..cfa4e19 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -81,6 +81,7 @@ dtb-$(CONFIG_DEVEL_BOARD) += armada-7040-rz-db_nand.dtb \
 			     armada-7040-rz-db_rxaui_pci.dtb \
 			     armada-7040-rz-db_rxaui.dtb
 else
+dtb-y += armada-70x0-customer0.dtb
 dtb-$(CONFIG_DEVEL_BOARD) += armada-70x0-db_router.dtb
 endif
 endif #CONFIG_TARGET_ARMADA_8K
diff --git a/arch/arm/dts/armada-70x0-customer0.dts b/arch/arm/dts/armada-70x0-customer0.dts
new file mode 100644
index 0000000..dfbea08
--- /dev/null
+++ b/arch/arm/dts/armada-70x0-customer0.dts
@@ -0,0 +1,175 @@
+/*
+ * ***************************************************************************
+ * Copyright (C) 2015 Marvell International Ltd.
+ * ***************************************************************************
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 2 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ***************************************************************************
+ */
+
+/dts-v1/;
+
+#include "armada-70x0.dtsi" /* include SoC device tree */
+
+/ {
+	model = "DB-ARMADA-70x0 (Customer)";
+	compatible = "marvell,armada-70x0";
+	fdt_config_id = <1>;
+	board_id = <0x11>;
+
+	ap-806 {
+		internal-regs {
+			spi0: spi@510600 {
+				status = "disabled";
+			};
+			i2c0: i2c@511000 {
+				status = "okay";
+			};
+			pinctl: pinctl@6F4000 {
+					/* MPP Bus:
+						SPI0 [0-3]
+						I2C0 [4-5]
+						UART0 [11,19]
+					*/
+					  /* 0 1 2 3 4 5 6 7 8 9 */
+				pin-func = < 3 3 3 3 3 3 0 0 0 0
+					     0 3 0 0 0 0 0 0 0 3>;
+			};
+		};
+	};
+	cp110-master {
+		internal-regs {
+			spi0@700600 {
+				status = "disabled";
+			};
+			spi1@700680 {
+				status = "okay";
+			};
+			i2c@701000 {
+				status = "disabled";
+			};
+			usb3@500000 {
+				status = "okay";
+			};
+			usb3@510000 {
+				status = "okay";
+			};
+			sata@540000 {
+				status = "okay";
+			};
+			mmc@780000 {
+				status = "disabled";
+			};
+			pcie-controller {
+				pcie@0,0 {
+					status = "disabled";
+				};
+				pcie@1,0 {
+					status = "disabled";
+				};
+				pcie@2,0 {
+					status = "okay";
+				};
+			};
+			nand@720000 {
+				status = "disabled";
+			};
+
+			pinctl@440000 {
+					/* MPP Bus:
+					   RGMII0[0-11]
+					   SPI   [13-16]
+					   SATA1 [28]
+					   UART0 [29-30]
+					   SMI	 [32,34]
+					   XSMI  [35-36]
+					   I2C	 [37-38]
+					   UART1 [40-41]
+					   RGMII1[44-55]
+					   SD	 [56-62]
+					*/
+					/*   0   1   2   3   4   5   6   7   8   9 */
+				pin-func = < 3   3   3   3   3   3   3   3   3   3
+					     3   3   0   3   3   3   3   0   0   0
+					     0   0   0   0   0   0   0   0   9   0xA
+					     0xA 0   7   0   7   7   7   2   2   0
+					     0   0   0   0   1   1   1   1   1   1
+					     1   1   1   1   1   1   0xE 0xE 0xE 0xE
+					     0xE 0xE 0xE>;
+			};
+
+			comphy {
+				phy0 {
+					phy-type = <PHY_TYPE_SGMII2>;
+					phy-speed = <PHY_SPEED_1_25G>;
+				};
+				phy1 {
+					phy-type = <PHY_TYPE_USB3_HOST0>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy2 {
+					phy-type = <PHY_TYPE_SGMII0>;
+					phy-speed = <PHY_SPEED_1_25G>;
+				};
+				phy3 {
+					phy-type = <PHY_TYPE_SATA1>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy4 {
+					phy-type = <PHY_TYPE_USB3_HOST1>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy5 {
+					phy-type = <PHY_TYPE_PEX2>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+			};
+			mdio {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				mac0 {
+					phy-mode = "sgmii"; /* lane-2 */
+					status = "okay";
+				};
+				mac2 {
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&phy2>;
+					status = "okay";
+				};
+				mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+					status = "okay";
+				};
+			};
+			ppv22@000000 {
+				eth0@010000 {
+					status = "okay";
+				};
+				eth1@020000 {
+					status = "okay";
+				};
+				eth2@030000 {
+					status = "okay";
+				};
+			};
+		};
+	};
+};
-- 
1.9.1

